leases/schema.py

```diff
@@ -92,6 +93,9 @@ def mutate_and_get_payload(cls, root, info, **input):
 
             input["boat"] = boat
 
+        application.status = ApplicationStatus.OFFER_GENERATED
```

Comment:
I was thinking that we could maybe enforce in the model's `clean` method, that it cannot be `PENDING` if it has a `BerthLease` related to it. Without the lease it can be `PENDING` and `EXPIRED`, but with the lease other statuses apply. 🤔

Reply:
Sounds like a good approach. 👍